authentication/http.go

```diff
@@ -30,7 +28,7 @@ const (
 // WithTenant finds the tenant from the URL parameters and adds it to the request context.
 func WithTenant(next http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		tenant := chi.URLParam(r, "tenant")
+		tenant := r.URL.Query().Get(tenantKey.String())
```

Comment:
Should we fail request if no tenant was provided?

Reply:
Also this key is weird ...`HTTP context key` as URL param 🤔 

![image](https://user-images.githubusercontent.com/6950331/123751041-2d8ca300-d8af-11eb-9082-ed0fdae8f295.png)

Reply:
removed all the references to add/retrieve tenant names to/from query-params